diff --git a/feeds.conf.default b/feeds.conf.default
index 212f8a0119..2b69565de5 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -5,6 +5,8 @@ src-git telephony https://git.openwrt.org/feed/telephony.git
 src-git nss_packages https://github.com/qosmio/nss-packages.git;NSS-12.5-K6.x
 src-git sqm_scripts_nss https://github.com/qosmio/sqm-scripts-nss.git
 src-git video https://github.com/openwrt/video.git
+src-git passwall https://github.com/xiaorouji/openwrt-passwall.git;main
+src-git passwall_packages https://github.com/xiaorouji/openwrt-passwall-packages.git;main
 #src-git targets https://github.com/openwrt/targets.git
 #src-git oldpackages http://git.openwrt.org/packages.git
 #src-link custom /usr/src/openwrt/custom-feed
diff --git a/target/linux/qualcommax/image/ipq60xx.mk b/target/linux/qualcommax/image/ipq60xx.mk
index b476cbfa84..c482b4c910 100644
--- a/target/linux/qualcommax/image/ipq60xx.mk
+++ b/target/linux/qualcommax/image/ipq60xx.mk
@@ -209,7 +209,6 @@ define Device/zn_m2
        PAGESIZE := 2048
        DEVICE_DTS_CONFIG := config@cp03-c1
        SOC := ipq6018
-       DEVICE_PACKAGES := ipq-wifi-zn_m2
 endef
 TARGET_DEVICES += zn_m2

diff --git a/target/linux/qualcommax/ipq60xx/target.mk b/target/linux/qualcommax/ipq60xx/target.mk
index dea833d3fe..d676fa4119 100644
--- a/target/linux/qualcommax/ipq60xx/target.mk
+++ b/target/linux/qualcommax/ipq60xx/target.mk
@@ -1,7 +1,7 @@
 SUBTARGET:=ipq60xx
 BOARDNAME:=Qualcomm Atheros IPQ60xx
 DEFAULT_PACKAGES += \
-       ath11k-firmware-ipq6018 nss-firmware-ipq6018 kmod-qca-nss-drv-bridge-mgr kmod-qca-nss-drv-clmapmgr kmod-qca-nss-drv-eogremgr kmod-qca-nss-drv-gre \
+    nss-firmware-ipq6018 kmod-qca-nss-drv-bridge-mgr kmod-qca-nss-drv-clmapmgr kmod-qca-nss-drv-eogremgr kmod-qca-nss-drv-gre \
        kmod-qca-nss-drv-igs kmod-qca-nss-drv-l2tpv2 kmod-qca-nss-drv-lag-mgr kmod-qca-nss-drv-map-t kmod-qca-mcs kmod-qca-nss-ecm \
        kmod-qca-nss-drv-match kmod-qca-nss-drv-mirror kmod-qca-nss-drv-pppoe kmod-qca-nss-drv-pptp kmod-qca-nss-drv-pvxlanmgr kmod-qca-nss-drv-qdisc \
        kmod-qca-nss-drv-tun6rd kmod-qca-nss-drv-tunipip6 kmod-qca-nss-drv-vlan-mgr kmod-qca-nss-drv-vxlanmgr kmod-qca-nss-macsec
